controller/work_item_link_category_blackbox_test.go

```diff
@@ -142,12 +146,91 @@ func (s *workItemLinkCategorySuite) createWorkItemLinkCategoryUser() (http.Respo
 	return test.CreateWorkItemLinkCategoryCreated(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, &payload)
 }
 
+func createWorkItemLinkCategorySystemInRepo(t *testing.T, db application.DB, ctx context.Context) uuid.UUID {
+	id := "0e671e36-871b-43a6-9166-0c4bd573e231"
+	return createWorkItemLinkCategoryInRepo(t, db, ctx, "test-system", "This work item link category is reserved for the core system.", &id)
+}
+
+func createWorkItemLinkCategoryUserInRepo(t *testing.T, db application.DB, ctx context.Context) uuid.UUID {
+	id := "bf30167a-9446-42de-82be-6b3815152051"
+	return createWorkItemLinkCategoryInRepo(t, db, ctx, "test-user", "This work item link category is managed by an admin user.", &id)
+}
+
+func createWorkItemLinkCategoryInRepo(t *testing.T, db application.DB, ctx context.Context, name string, description string, id *string) uuid.UUID {
+	var cID uuid.UUID
+	if id == nil {
+		cID = uuid.NewV4()
+	} else {
+		cID = uuid.FromStringOrNil(*id)
+	}
+
+	// Use the goa generated code to create a work item link category
+	payload := app.CreateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
+			ID:   &cID,
+			Type: link.EndpointWorkItemLinkCategories,
+			Attributes: &app.WorkItemLinkCategoryAttributes{
+				Name:        &name,
+				Description: &description,
+			},
+		},
+	}
+
+	err := application.Transactional(db, func(appl application.Application) error {
+		modelCategory := ConvertLinkCategoryToModel(app.WorkItemLinkCategorySingle{Data: payload.Data})
+		_, err := appl.WorkItemLinkCategories().Create(ctx, &modelCategory)
+		return err
+	})
+	require.Nil(t, err)
+	return cID
+}
+
 //-----------------------------------------------------------------------------
 // Actual tests
 //-----------------------------------------------------------------------------
 
+func (s *workItemLinkCategorySuite) TestCreateAndDeleteWorkItemLinkCategoryFails() {
+	name := "test-user"
+	description := "This work item link category is managed by an admin user."
+	id := uuid.FromStringOrNil("bf30167a-9446-42de-82be-6b3815152051")
+
+	// Use the goa generated code to create a work item link category
+	payload := app.CreateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
+			ID:   &id,
+			Type: link.EndpointWorkItemLinkCategories,
+			Attributes: &app.WorkItemLinkCategoryAttributes{
+				Name:        &name,
+				Description: &description,
+			},
+		},
+	}
+
+	test.CreateWorkItemLinkCategoryMethodNotAllowed(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, &payload)
+	test.DeleteWorkItemLinkCategoryMethodNotAllowed(s.T(), s.svc.Context, s.svc, s.linkCatCtrl, id)
+}
+
+func (s *workItemLinkCategorySuite) TestUpdateWorkItemLinkCategoryFails() {
+	name := "Some name"
+	description := "New description for work item link category."
+	id := uuid.FromStringOrNil("88727441-4a21-4b35-aabe-007f8273cd19")
+	payload := &app.UpdateWorkItemLinkCategoryPayload{
+		Data: &app.WorkItemLinkCategoryData{
```

Comment:
Done in https://github.com/fabric8-services/fabric8-wit/pull/1521/commits/20c8e3b1da0c48138f6a15728a67296c6a0b33bc